src/libraries/Microsoft.PowerFx.Interpreter/Utility.cs

```diff
@@ -30,5 +35,34 @@ public static T GetService<T>(this IServiceProvider serviceProvider)
         {
             return (T)serviceProvider.GetService(typeof(T));
         }
+
+        public static FormulaValue SetUntypedObject(this UntypedObjectBase untypedObject, IRContext context, FormulaValue property, FormulaValue value)
+        {
+            try
+            {
+                if (property is StringValue stringValue)
+                {
+                    untypedObject.SetProperty(stringValue.Value, value);
+                }
+                else if (property is NumberValue numberValue)
+                {
+                    untypedObject.SetProperty((int)numberValue.Value, value);
+                }
```

Comment:
We should avoid switching logic like this.  It makes it too tempting to allow wrong behavior. 

Safest to just have 2 separate methods. 

Or if we're trying to share wrapping, we can use a callback, 
SetUntypedAction(...,  Action worker) 

where worker can either be Set by nam, or set by index.